### PR TITLE
 Implements authorization for ConnectedAasManager

### PR DESCRIPTION
**Pull request by mdanish98**
The current implementation of [ConnectedAasManager](https://github.com/eclipse-basyx/basyx-java-server-sdk/blob/main/basyx.aasenvironment/basyx.aasenvironment-client/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/client/ConnectedAasManager.java) should be extended to enable authorized requests. 

This PR implements authorized versions of below Connected components:
- Implements authorization for ConnectedAasManager
- Implements authorization for ConnectedAasRepository
- Implements authorization for ConnectedAasService
- Implements authorization for ConnectedSubmodelRepository
- Implements authorization for ConnectedSubmodelService

Signed-off-by: Mohammad Ghazanfar Ali Danish <ghazanfar.danish@iese.fraunhofer.de>

### CODE REVIEW

File: basyx.aasregistry/basyx.aasregistry-service-release-kafka-mongodb/src/test/java/org/eclipse/digitaltwin/basyx/aasregistry/service/storage/mongodb/AuthorizedClientTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+			registryApi.postAssetAdministrationShellDescriptor(descriptor);
+		});
+
+		assertEquals(HttpStatus.UNAUTHORIZED.value(), exception.getCode());
+	}
+	
+	@Test
+	@Override
+	public void whenPostSubmodelDescriptor_LocationIsReturned() throws ApiException, IOException {
+		// TODO: It uses normal GET unauthorized request, need to override and refactor
```
Comment (mateusmolina-iese) [inline basyx.aasregistry/basyx.aasregistry-service-release-kafka-mongodb/src/test/java/org/eclipse/digitaltwin/basyx/aasregistry/service/storage/mongodb/AuthorizedClientTest.java:104]: Are you implementing in the scope of the PR?

File: basyx.aasregistry/basyx.aasregistry-service-release-kafka-mongodb/src/test/java/org/eclipse/digitaltwin/basyx/aasregistry/service/storage/mongodb/AuthorizedClientTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	}
+	
+	@Test
+	@Override
+	public void whenPostShellDescriptor_LocationIsReturned() throws ApiException, IOException {
+		// TODO: It uses normal GET unauthorized request, need to override and refactor
```
Comment (mateusmolina-iese) [inline basyx.aasregistry/basyx.aasregistry-service-release-kafka-mongodb/src/test/java/org/eclipse/digitaltwin/basyx/aasregistry/service/storage/mongodb/AuthorizedClientTest.java:110]: Are you implementing in the scope of the PR?

File: basyx.submodelregistry/basyx.submodelregistry-service-release-kafka-mongodb/src/test/java/org/eclipse/digitaltwin/basyx/submodelregistry/service/storage/mongodb/AuthorizedClientTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+			registryApi.postSubmodelDescriptor(descriptor);
+		});
+
+		assertEquals(HttpStatus.UNAUTHORIZED.value(), exception.getCode());
+	}
+	
+	@Test
+	@Override
+	public void whenPostSubmodelDescriptor_LocationIsReturned() throws ApiException, IOException {
+		// TODO: It uses normal GET unauthorized request, need to override and refactor
```
Comment (mateusmolina-iese) [inline basyx.submodelregistry/basyx.submodelregistry-service-release-kafka-mongodb/src/test/java/org/eclipse/digitaltwin/basyx/submodelregistry/service/storage/mongodb/AuthorizedClientTest.java:104]: Are you implementing in the scope of the PR?

File: basyx.submodelservice/basyx.submodelservice-client/pom.xml [hunk 4]
```diff
@@ -55,5 +55,48 @@
...
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.authorization</artifactId>
+			<scope>test</scope>
+			<classifier>tests</classifier>
+		</dependency>
+		
+		<dependency>
+			<groupId>org.eclipse.digitaltwin.basyx</groupId>
+			<artifactId>basyx.submodelrepository-feature-authorization</artifactId>
+			<scope>test</scope>
```
Comment (mateusmolina-iese) [inline basyx.submodelservice/basyx.submodelservice-client/pom.xml:86]: using this pom as reference, but applies to all changed poms: did you run mvn dependency:analyze to see if all the added deps are used?

**Review by mateusmolina-iese — CHANGES_REQUESTED** [review]

File: ci/docker-compose.yml [hunk 5]
```diff
@@ -109,6 +149,38 @@ services:
       networks:
         - basyx-java-server-sdk
 
+  keycloak-fixed-uri:
+    image: eclipsebasyx/keycloak:0.0.1
+    build:
+      context: ./../examples/keycloak
+      dockerfile_inline: |
```
[the remark this answers is not in the document]
Comment (mdanish98) [reply ci/docker-compose.yml:156]: Done

File: examples/keycloak/initializer/src/main/java/org/eclipse/digitaltwin/basyx/keycloak/initializer/InitializerSpi.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package org.eclipse.digitaltwin.basyx.keycloak.initializer;
+
+import com.google.auto.service.AutoService;
+import org.keycloak.provider.Provider;
+import org.keycloak.provider.ProviderFactory;
+import org.keycloak.provider.Spi;
+
+@AutoService(Spi.class)
+public class InitializerSpi implements Spi {
```
[the remark this answers is not in the document]
Comment (mdanish98) [reply examples/keycloak/initializer/src/main/java/org/eclipse/digitaltwin/basyx/keycloak/initializer/InitializerSpi.java:9]: Done

File: basyx.aasregistry/basyx.aasregistry-client-native/pom.xml [hunk 7]
```diff
@@ -19,8 +19,24 @@
 		<openapi.result.file>${openapi.result.folder}/${openapi.name}</openapi.result.file>
 	</properties>
 	<build>
-		<sourceDirectory>src/generated/java</sourceDirectory>
 		<plugins>
+			<plugin>
```
[the remark this answers is not in the document]
Comment (mdanish98) [reply basyx.aasregistry/basyx.aasregistry-client-native/pom.xml:23]: Removed

File: basyx.submodelregistry/basyx.submodelregistry-client-native/hs_err_pid14700.log
[the remark this answers is not in the document]
Comment (mdanish98) [reply basyx.submodelregistry/basyx.submodelregistry-client-native/hs_err_pid14700.log:1]: Done

File: basyx.submodelregistry/basyx.submodelregistry-client-native/hs_err_pid70168.log
[the remark this answers is not in the document]
Comment (mdanish98) [reply basyx.submodelregistry/basyx.submodelregistry-client-native/hs_err_pid70168.log:1]: Done

File: basyx.aasregistry/basyx.aasregistry-service-release-log-mem/hs_err_pid21812.log
[the remark this answers is not in the document]
Comment (mdanish98) [reply basyx.aasregistry/basyx.aasregistry-service-release-log-mem/hs_err_pid21812.log:1]: Done

File: basyx.aasregistry/basyx.aasregistry-service-release-log-mem/hs_err_pid68424.log
[the remark this answers is not in the document]
Comment (mdanish98) [reply basyx.aasregistry/basyx.aasregistry-service-release-log-mem/hs_err_pid68424.log:1]: Done

**Review by mateusmolina-iese — APPROVED** [review]

**Pull request merged**